manifests/platforms/eks.jsonnet

```diff
@@ -0,0 +1,147 @@
+
+{
+  config:: error "no kubeprod configuration",
+
+  // Shared metadata for all components
+  kubeprod: kube.Namespace("kubeprod"),
+
+  external_dns_zone_name:: $.config.dnsZone,
+  letsencrypt_contact_email:: $.config.contactEmail,
+  letsencrypt_environment:: "prod",
+
+  version: version,
+
+  grafana: grafana {
+    prometheus:: $.prometheus.prometheus.svc,
+    ingress+: {
+      host: "grafana." + $.external_dns_zone_name,
+    },
+  },
+
+  edns: edns {
+    local this = self,
+
+    // NOTE: https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-credentials
+    // for additional information on how to use environment variables to configure a particular user when accessing
+    // the AWS API.
+    secret: kube.Secret(this.p + "external-dns-aws-conf") {
+      metadata+: {
+        namespace: "kubeprod",
+      },
+      data_+: $.config.externalDns,
+    },
+
+    deploy+: {
+      ownerId: $.external_dns_zone_name,
+      spec+: {
+        template+: {
+          spec+: {
+            containers_+: {
+              edns+: {
+                env_+: {
+                  AWS_ACCESS_KEY_ID: kube.SecretKeyRef(this.secret, "aws_access_key_id"),
+                  AWS_SECRET_ACCESS_KEY: kube.SecretKeyRef(this.secret, "aws_access_key_secret"),
+                },
+                args_+: {
+                  provider: "aws",
+                  "aws-zone-type": "public",
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  },
+
+  cert_manager: cert_manager {
+    letsencrypt_contact_email:: $.letsencrypt_contact_email,
+    letsencrypt_environment:: $.letsencrypt_environment,
+  },
+
+  nginx_ingress: nginx_ingress,
+
+  oauth2_proxy: oauth2_proxy {
+    secret+: {
+      data_+: $.config.oauthProxy,
+    },
+
+    deploy+: {
+      spec+: {
+        template+: {
+          spec+: {
+            containers_+: {
+              proxy+: {
+                args_+: {
+                  provider: "oidc",
+                  "oidc-issuer-url": "https://cognito-idp.%s.amazonaws.com/%s" % [
+                    $.config.oauthProxy.aws_region,
+                    $.config.oauthProxy.aws_user_pool_id
+                  ],
+                  /* NOTE: disable cookie refresh token.
+                   * As per https://docs.aws.amazon.com/cognito/latest/developerguide/token-endpoint.html:
+                   * The refresh token is defined in the specification, but is not currently implemented to
+                   * be returned from the Token Endpoint.
+                   */
+                  "cookie-refresh": "0",
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  },
+
+  prometheus: prometheus {
+    ingress+: {
+      host: "prometheus." + $.external_dns_zone_name,
+    },
+  },
+
+  fluentd_es: fluentd_es {
+    es:: $.elasticsearch,
+  },
+
+  elasticsearch: elasticsearch {
```

Comment:
Minor: suggest removing `{\n}` no-op (unless you want/need it for some specific reason).